### PR TITLE
#164139687 Add ratings to an article

### PR DESCRIPTION
**Pull request by kleva-j**
#### What does this PR do?
Adds the feature for users to be able to rate an article.
#### Description of Task to be completed?
Users should be able to rate an article.

This new endpoint takes these input:
- `rating` from the `req.body` object
- `articleId` from the `req.param` object
#### How should this be manually tested?
- run  `npm run db-migrate` to get all database migration updates.
- make a 'POST' request to the `/api/v1/articles/rate/:articleId` route.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139687

#### Questions:
N/A

### CODE REVIEW

File: controllers/articles.js [hunk 1]
```diff
@@ -31,7 +31,56 @@ export default class ArticleController {
...
+ * @param {Object} res - the response object
+ * @memberof ArticleController
+ * @returns {Object} class instance
+ */
+  static async rateArticle(req, res) {
+    const { id } = req.user;
+    const rating = Number(req.body.rating);
+    const { articleId } = req.params;
+    try {
+      const articleExist = await Article.findOne({ where: { id: articleId } });
```
Comment (olajide1234) [inline controllers/articles.js]: Try using middleware to find if the article exists and if the user owns the article. Let's sync.

File: controllers/articles.js [hunk 2]
```diff
@@ -31,7 +31,56 @@ export default class ArticleController {
...
+            return res.status(201).json({
+              success: true,
+              message: `Article rating has been updated as ${rating}`,
+              rating: updatedRating
+            });
+          }
+          return res.status(200).json({
+            success: true,
+            message: `Article has been rated as ${rating}`,
+            articleRating: (await Ratings.create({
```
Comment (olajide1234) [inline controllers/articles.js]: Slick! However, please can you consider extracting create rating into a separate function? It makes the code more understandable and easier to maintain. Thanks.

**Review by olajide1234 — COMMENTED** [review]
Great job. Please look at these few comments.

**Review by codinger41 — APPROVED** [review]

**Review by youngestdj — COMMENTED** [review]
Fix hound errors and you'd be good to go

**Review by youngestdj — APPROVED** [review]

File: middleware/checkIfArticleExist.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+import { Article } from '../models';
```
Comment (9jaswag) [inline middleware/checkIfArticleExist.js:2]: @kleva-j this file can be named something better

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: doc.json [hunk 4]
```diff
@@ -231,125 +231,56 @@
...
-        "operationId": "CategoryPost",
-        "deprecated": false
-      }
-    },
-    "/requestpasswordreset": {
+    "/articles": {
       "post": {
         "tags": [
-          "Reset password"
+          "Articles"
```
Comment (9jaswag) [inline doc.json:237]: @kleva-j why all these changes here?

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**